src/searcher.js

```diff
@@ -0,0 +1,61 @@
+import { LitElement, html } from "lit-element";
+
+export class SearcherComp extends LitElement {
+  static get properties() {
+    return {
+      cover: { searcher: Object },
+      result: { type: Array },
+      data: { type: Array },
+    };
+  }
+
+  constructor() {
+    super();
+    this.data = [];
+    this.result = this.data;
+  }
+
+  sendData(data) {
+    this.dispatchEvent(
+      new CustomEvent("datasave", {
+        detail: { data },
+        bubbles: true,
+        composed: true,
+      })
+    );
+  }
+
+  render() {
+    return html`
+      <div class="lupita">
+        <div class="input">
+          <input @input=${this.getText} size="100" type="text" name="searcher" 
+                  id="searcher" placeholder="Buscar por Titulo de Libros"/>
+        </div>
+      </div>
+    `;
+  }
+
+  getText(event) {
+    const input = event.target;
+    this.filterData(input.value);
+  }
+
+  filterData(input) {
+    console.log(input, this.result);
+    this.result = [];
+    this.data.filter((product) => {
+      const name = product.book_title.toLowerCase();
+
+      if (name.indexOf(input) !== -1) this.result = [...this.result, product];
+    });
```

Comment:
the `filter` function returns a new array with the filtered data, check this [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)